test/widgets/lightbox_test.dart

```diff
@@ -197,6 +204,128 @@ class FakeVideoPlayerPlatform extends Fake
+      await tester.tap(messageListImageFinder);
+      await tester.pump();
+      // pump to start hero animation
+      await tester.pump();
+
+      final heroAnimationDuration = Duration(milliseconds: 300);
+      final steps = 150;
+      final stepDuration = heroAnimationDuration ~/ steps;
+      List<Rect> animatedPositions = [];
+      for (int i = 1; i <= steps; i++) {
+        await tester.pump(stepDuration);
+
+        final animatedFlightImageFinder = find.byWidgetPredicate((widget) =>
+          widget is RealmContentNetworkImage && widget.src == imageSrcUrl
+        );
+        animatedPositions.add(tester.getRect(animatedFlightImageFinder));
+      }
+
+      final totalDistance = sqrt(
+        pow(initialImagePosition.top - animatedPositions.last.top, 2) +
+        pow(initialImagePosition.left - animatedPositions.last.left, 2));
+
+      Rect previousPosition = initialImagePosition;
+      double maxStepDistance = 0.0;
+      for (final position in animatedPositions) {
+        final stepDistance = sqrt(
+          pow(position.top - previousPosition.top, 2) +
+          pow(position.left - previousPosition.left, 2));
+        maxStepDistance = max(maxStepDistance, stepDistance);
+        check(position).not((pos) => pos.equals(previousPosition));
+
+        previousPosition = position;
+      }
+      check(maxStepDistance).isLessThan(0.03 * totalDistance);
+
+      debugNetworkImageHttpClientProvider = null;
+    });
+
+    testWidgets('no hero animation occurs between different message list pages for same image', (tester) async {
+      prepareBoringImageHttpClient();
+
+      await setupMessageListPage(tester);
+
+      final imageFinder = find.byWidgetPredicate((widget) =>
+        widget is RealmContentNetworkImage && widget.src == imageSrcUrl
+      );
+      final firstImagePosition = tester.getRect(imageFinder);
+      final firstElement = imageFinder.evaluate().single;
+
+      connection.prepare(json:
+        eg.newestGetMessagesResult(foundOldest: true, messages: [message]).toJson());
+      await tester.tap(find.descendant(
+        of: find.byType(StreamMessageRecipientHeader),
+        matching: find.text('test topic')));
+      await tester.pumpAndSettle();
+
+      final secondImagePosition = tester.getRect(imageFinder);
+      final secondElement = imageFinder.evaluate().single;
+
+      await tester.tap(find.byType(BackButton));
+      await tester.pump();
+
+      final heroAnimationDuration = Duration(milliseconds: 300);
+      final steps = 150;
+      final stepDuration = heroAnimationDuration ~/ steps;
+      for (int i = 0; i < steps; i++) {
+        await tester.pump(stepDuration);
+
+        final currentImages = imageFinder.evaluate();
+        check(currentImages).unorderedEquals([firstElement, secondElement]);
+
+        check(tester.getRect(
+          find.byElementPredicate((e) => e == firstElement))).equals(firstImagePosition);
```

Comment:
nit: line too long; instead:
```suggestion
        check(tester.getRect(find.byElementPredicate((e) => e == firstElement)))
          .equals(firstImagePosition);
```

Reply:
Better yet, make a little helper function within this test case, and then:
```suggestion
        check(getElementRect(firstElement)).equals(firstImagePosition);
```

Then the same `getElementRect` can be used for defining `firstImagePosition` in terms of `firstElement` in the first place. That way this check works the same way as `firstImagePosition` got computed originally — so we're doing the same thing the same way both times, instead of two somewhat different ways, and that means one fewer thing for the reader to understand and think through.